File: 2021/csharp/Aoc2021.Tests/Day00Tests.cs
namespace Aoc2021.Tests;

public class Day00Tests
{
    [Theory]
    [InlineData(2, 12)]
    [InlineData(2, 14)]
    [InlineData(654, 1969)]
    [InlineData(33583, 100756)]
    public void RequiredFuelWorks(int fuel, int mass)
    {
        Assert.Equal(fuel, Day00.RequiredFuel(mass));
    }

    [Theory]
    [InlineData(2, 14)]
    [InlineData(966, 1969)]
    [InlineData(50346, 100756)]
    public void RequiredTotalFuelWorks(int fuel, int mass)
    {
        Assert.Equal(fuel, Day00.RequiredTotalFuel(mass));
    }

    [Fact]
    public void Stars()
    {
        Day00 run = new();
        Assert.Equal("3421505", run.Result1());
        Assert.Equal("5129386", run.Result2());
    }
}

File: 2021/fsharp/Aoc2021.Tests/Day00.Tests.fs
module Day00.Tests

open Xunit
open Day00

[<Theory>]
[<InlineData(12, 2)>]
[<InlineData(14, 2)>]
[<InlineData(1969, 654)>]
[<InlineData(100756, 33583)>]
let ``Day00 fuel works`` mass expectedFuel = Assert.Equal(expectedFuel, fuel mass)

[<Fact>]
let ``Day00 Stars`` () =
    Assert.Equal("3421505", job1 ())
    Assert.Equal("5129386", job2 ())
